src/pilot/assign_role.py

```diff
@@ -381,22 +385,31 @@ def get_raid_controller_physical_disk_ids(drac_client, raid_controller_fqdd):
         key=physical_disk_id_to_key)
 
 
-def define_storage_logical_disks(drac_client, raid_controller_name):
+def define_storage_logical_disks(drac_client, raid_controller_ids):
+    raid_controllers = []
     all_physical_disks = drac_client.list_physical_disks()
 
+    if len(raid_controller_ids) > 1:
+        for cntrl in raid_controller_ids:
+            raid_controllers.append(cntrl)
+    else:
+        raid_controllers.append(raid_controller_ids[0])
+
     # Get the drives controlled by the RAID controller
-    raid_cntlr_physical_disks = []
+    raid_cntlr_physical_disks = {}
     for disk in all_physical_disks:
-        if disk.controller == raid_controller_name:
-            raid_cntlr_physical_disks.append(disk)
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
+                raid_cntlr_physical_disks[disk.controller].append(disk)
+            else:
+                raid_cntlr_physical_disks[disk.controller] = [disk]
 
     # Make sure we have enough drives attached to the RAID controller to create
     # a RAID1
-    num_raid_cntlr_physical_disks = len(raid_cntlr_physical_disks)
-    if num_raid_cntlr_physical_disks < 2:
+    if all(len(raid_cntlr_physical_disks[cntrl]) < 2
```

Comment:
It's really not necessary for all RAID controllers to have 2 or more disks.  We really just need 1 RAID controller to have 2 or more.  Let's break this code (lines 409 through 413) out into a method that returns True or False and then call that method after we know which controller we're using.  So that would be immediately after line 425 and immediately after line 429.

Reply:
Added method to check disk count.